### PR TITLE
Fix search to properly populate user data

### PR DESCRIPTION
**Pull request by yuna878**
## Overview

* Previously made search accept requests without auth headers
* Made a mistake so requests with auth headers had incorrect info
* Fixed the mistake!

## Changes Made

When request comes in, if auth header is not present, bypass. 
If auth header is present, verify the session token and throw an exception if needed

### CODE REVIEW

File: src/app/coursegrab/utils/authorize.py [hunk 1]
```diff
@@ -26,3 +26,19 @@ def inner(*args, **kwargs):
...
+    If auth header is present, verify the session.
+    """
+    @wraps(f)
+    def inner(*args, **kwargs):
+        auth_header = request.headers.get("Authorization")
+        if auth_header is None:
+            return f(*args, **kwargs)
+        bearer_token = auth_header.replace("Bearer ", "").strip()
+        user, session = verify_session(bearer_token)
+        return f(user=user, session=session, *args, **kwargs)
```
Comment (alanna-zhou) [inline src/app/coursegrab/utils/authorize.py:42]: why do you need to have a wrap thingie?

**Review by alanna-zhou — APPROVED** [review]
seems fine but not sure what the `@wraps(f)` thing does haha

**Review by katesliang — APPROVED** [review]
looks great!!!!! 👍 i too am confused what wraps is tho

**Review by tedim52 — APPROVED** [review]
LGTM : )

File: src/app/coursegrab/utils/authorize.py (hunk 1 above)
Comment (yuna878) [reply src/app/coursegrab/utils/authorize.py:42]: wraps() basically allows the wrapper function (in this case, `inner`) to have the same property as the wrapped function (in this case, `f`) . Not super necessary, but helps with debugging and it's a common practice!

**Pull request merged**